### PR TITLE
Adding details to NTXentLoss documentation

### PR DESCRIPTION
**Pull request by stompsjo**
Hello! I use this repo so much I figured I ought to contribute where I can. I use `NTXentLoss` quite often and figured I could take a try at adding some documentation and address Issue #608.

I drew largely from @KevinMusgrave's [comment](https://github.com/KevinMusgrave/pytorch-metric-learning/issues/179#issuecomment-1503822592) to create an example of how embeddings are handled by the loss function. I realize much of the information I added is probably not specific enough to this loss function, so feel free to trim and cut to make it more concise.

(You can also scrap this PR entirely if it isn't what you had in mind for #608! I figured so many people are interested in temperature normalized cross-entropy loss that some additional documentation would benefit us all 😄)

### CODE REVIEW

**Comment by KevinMusgrave** [comment]
Thanks @stompsjo this is great! I made some edits and merged the changes. You can see the updated docs: https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#ntxentloss. The explanation is in the expandable section "How exactly is the NTXentLoss computed?"
